agreement_helpdesk_mgmt_sale/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    agreement_id = fields.Many2one(
+        comodel_name="agreement",
+        compute="_compute_agreement_id",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("sale_order_ids.agreement_id")
+    def _compute_agreement_id(self):
+        for ticket in self:
+            agreements = ticket.mapped("sale_order_ids.agreement_id")
+            if not agreements:
+                ticket.agreement_id = False
+            elif len(agreements) == 1:
+                ticket.agreement_id = agreements.id
+            else:
+                ticket.agreement_id = False
```

Comment:
```suggestion
        if hasattr(super(), "_compute_agreement_id"):
            super()._compute_agreement_id()
        for ticket in self:
            if not ticket.agreement_id and len(ticket.sale_order_ids.agreement_id) == 1:
                ticket.agreement_id = ticket.sale_order_ids.agreement_id
```


Since the field was created by another module, and this one's only making it computed.. better to play nice with other modules potentially doing the same, so:

1. Consider that `_compute_agreement_id` may be defined by another module
2. Consider that agreement_id may have already been set, and set only when needed.